conda/core/subdir_data.py

```diff
@@ -80,6 +79,22 @@ def __call__(cls, channel, repodata_fn=REPODATA_FN):
         return subdir_data_instance
 
 
+class LazyPackageRecords(UserList):
```

Comment:
In the spirit of naming things is hard, let's focus on the positive and just name this:

```suggestion
class PackageRecordList(UserList):
```